Shim script generator and vscode instructions

wadetandy opened the pull request:
Reopening from #201 after the rebase.  This adds a new script for generating an executable shim script as opposed to the currently recommended alias, as well as README instructions for setting up vscode. Does not include README update for the shim script, but glad to update with that if you want to make it the path going forward.

wadetandy committed:
Add instructions for setting up vscode integration

wadetandy committed:
Add script for generating a shim rubyfmt script

The README currently recommends setting up a `rubyfmt` alias as part of
the installation process, but depending on how you are invoking the
command, not every environment will work with an alias.  This script
will install a shim script at the location of the user's choice:

``` bash
$ ./script/install_shim.sh ~/bin/rubyfmt
```

The script adds the `RUBYFMT_USE_RELEASE` variable as well so as to
prevent unnecessary debug output.

wadetandy committed:
Add test for input via stdin

Originally wrote this test when trying to add the feature, but when I
pulled the latest changes, the CLI had already added the functionality,
so I figured I'd contribute the test back to it at least.

fables-tales commented on script/install_shim.sh:
```diff
@@ -0,0 +1,12 @@
+echo "#!/usr/bin/env bash
+export RUBYFMT_USE_RELEASE=1
+$(which ruby) --disable=all $(pwd)/rubyfmt.rb \$@" > "$1"
+chmod +x "$1"
```
could you add a trailing newline here?

fables-tales commented on script/install_shim.sh:
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
could you add a test for this shim script?

fables-tales reviewed (changes requested):
mostly lgtm, just a few requests.

wadetandy committed:
Add test for shim script generator script

Also improved script so that it can run correctly when called from
anywhere, not just rubyfmt root.

wadetandy commented:
@penelopezone Updated with requested changes

fables-tales commented on README.md:
```diff
@@ -52,3 +52,17 @@ e.g.:
+## Editor Support
+
+### Visual Studio Code
+
```
you should reference the shim script, and then I will merge this.

wadetandy replied:
Updated README ✅

wadetandy committed:
Improve README with executable shim instructions
